sys/include/auto_init.h

```diff
@@ -1,35 +1,72 @@
 /*
  * Copyright (C) 2010,2015 Freie Universität Berlin
  * Copyright (C) 2010 Kaspar Schleiser <kaspar@schleiser.de>
- * Copyright (C) 2013 INRIA
+ * Copyright (C) 2013-2018 Inria
  *
  * This file is subject to the terms and conditions of the GNU Lesser
  * General Public License v2.1. See the file LICENSE in the top level
  * directory for more details.
  */
 
 /**
- * @defgroup    sys_auto_init Auto-init
+ * @defgroup    sys_auto_init Auto-initialization
  * @ingroup     sys
  * @brief       Auto initialize modules
- * @note        This feature can be used by any application by adding auto_init to
- *              USEMODULE in the application's Makefile. auto_init will initialize
- *              any other included module that does not require a parameter in
- *              its init function, i.e. if the prototype looks like this: void
- *              MODULE_init(void). Most timer modules or simple drivers can be
- *              initialized by auto_init. The modules will be initialized in
- *              the context of the main thread right before the main function
- *              gets called. Be aware that most modules expect to be
- *              initialized only once, so do not call a module's init function
- *              when using auto_init unless you know what you're doing.
+ *
+ * This feature can be enabled in any application by adding the `auto_init`
+ * module to the application's `Makefile`:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~ {.mk}
+ * USEMODULE += auto_init
+ * ~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * `auto_init` initializes any included module that provides
+ * auto-initialization capabilities.
+ *
+ * Drivers or cpu peripherals that provides a @ref drivers_saul adaption and
+ * @ref drivers_netdev can be initialized automatically using the `auto_init`
+ * module.
+ *
+ * For regular device drivers such as sensors, actuators, network devices
+ * (@ref drivers), the default initialization parameters are taken from the
+ * `DRIVER_params.h` file provided by the driver implementation.
+ *
+ * The default initialization parameters can be overriden by the application in
+ * several ways (examples with the @ref drivers_bmp180 oversampling parameter
+ * `BMP180_PARAM_OVERSAMPLING`):
```

Comment:
I think all three paragraphs above should be merged, since they are independent from the very first paragraph, but belong thematically very close together. Also, I think the first one of these three is hard to read in rendered docs, due to using auto-refs. Please consider the following

```markdown
Device drivers that provides a [SAUL](@ref drivers_saul) adaptation and
[network device drivers](@ref drivers_netdev) can be initialized
automatically using by the `auto_init` module. Their default initialization
parameters need to be in a `DRIVER_params.h` file provided by the driver
implementation. They can be overriden by the application in several ways
(examples below with the @ref drivers_bmp180's oversampling parameter
`BMP180_PARAM_OVERSAMPLING`):
```